code/lighting/lighting_profiles.cpp

```diff
@@ -0,0 +1,166 @@
+		r = tnm_Uncharted;
+	}
+	if(name == "linear"){
+		r = tnm_Invalid;
+	}
+	if(name == "aces"){
+		r = tnm_Invalid;
+	}
+	if(name == "aces approximate"){
+		r = tnm_Invalid;
+	}
+	if(name == "cineon"){
+		r = tnm_Invalid;
+	}
+	if(name == "cineon"){
+		r = tnm_Invalid;
+	}
+	if(name == "reinhard jodie"){
+		r = tnm_Invalid;
+	}
+	if(name == "reinhard extended"){
+		r = tnm_Invalid;
+	}
+	if(name == "PPC"){
+		r = tnm_Invalid;
+	}
+	if(name == "PPC RGB"){
+		r = tnm_Invalid;
+	}
+	return r;
+}
+
+lighting_profile lighting_profile::default_profile;
+
+void lighting_profile::load_profiles()
+{
+	parse_all();
+}
+
+//The logic for grabbing all the parseable files
+
+void lighting_profile::parse_all()
+{
+	default_profile.reset();
+	if (cf_exists_full("lighting_profiles.tbl", CF_TYPE_TABLES)){
+		mprintf(("TABLES:Starting parse of lighting profiles.tbl"));
+		lighting_profile::parse_file("lighting_profiles.tbl");
+	}
+
+	mprintf(("TBM  =>  Starting parse of lighting profiles ...\n"));
+	parse_modular_table("*-ltp.tbm", lighting_profile::parse_file);
+}
+
+void lighting_profile::parse_default_section()
+{
+	bool keep_going = true;
+	SCP_string buffer;
+	TonemapperAlgorithm tn;
+	while(!optional_string("#END DEFAULT PROFILE") && keep_going){
+		keep_going = false;
+		if(optional_string("$Tonemapper:")){
+			stuff_string(buffer,F_NAME);
+			tn = lighting_profile::name_to_tonemapper(buffer);
+			default_profile.tonemapper = tn;
+			keep_going = true;
+		}
+		keep_going |= optional_parse_into_float("$PPC Toe Strength:",&default_profile.ppc_values.toe_strength);
+		keep_going |= optional_parse_into_float("$PPC Toe Length:",&default_profile.ppc_values.toe_length);
+		keep_going |= optional_parse_into_float("$PPC Shoulder Length:",&default_profile.ppc_values.shoulder_length);
+		keep_going |= optional_parse_into_float("$PPC Shoulder Strength:",&default_profile.ppc_values.shoulder_strength);
+		keep_going |= optional_parse_into_float("$PPC Toe Strength:",&default_profile.ppc_values.shoulder_angle);
+		keep_going |= optional_parse_into_float("$Exposure:",&default_profile.exposure);
+		//TODO: Handle case when there's no line matched but we haven't hit an #end
+		Assert(keep_going);
+	}
+}
+
+//Handle an individual file.
+void lighting_profile::parse_file(const char *filename)
+{
+	try
+	{
+		if (filename == nullptr){
+			//Normal parsing uses this:
+			//read_file_text_from_default(defaults_get_file("lighting_profiles.tbl"));
+			//To set the defaults, I intend to set them as virtual table lines before here
+			//so, if no file, gtfo.
+			return;
+		}
+		read_file_text(filename, CF_TYPE_TABLES);
+		reset_parse();
+		bool keep_going = true;
+		while(check_for_eof()==0 && keep_going){
+			if(optional_string("#DEFAULT PROFILE")){
+				keep_going = true;
+				parse_default_section();
+			}
+			//TODO: Handle case when there's no line matched but we haven't hit an #end
+			if(!keep_going)
+			{
+				advance_to_eoln(nullptr);
+				keep_going = true;
+			}
```

Comment:
This looks like it could cause problems:
keep_going is never set to false, so when you have a table that has an additional newline or something after the default profile end, you'll never advance_to_eoln and never reach the end of the file.
Seems like keep_going should be set to false at the beginning of the loop.
Though in all honesty, I'd just remove the loop here as well, and make it into a single ``if(optional_string("#DEFAULT PROFILE"))``, eventually possibly followed by a single ``if(optional_string("#SELECTABLE PROFILE")`` or whatever. There's no need for the overarching sections to exist more than once, especially for the default profile section.

Reply:
To the latter, looping allows for the sections to be in any order. That said, given there's only one section currently it can probably go from this version.